ccm_web/server/src/api/api.service.ts

```diff
@@ -21,4 +29,42 @@ export class APIService {
       }
     }
   }
+
+  async getCourseName (userLoginId: string, courseId: number): Promise<string | null> {
+    const requestor = await this.canvasService.createRequestorForUser(userLoginId)
+    let name = null
+    try {
+      const response = await requestor.get<Course>(`courses/${courseId}`)
+      const course = response.body
+      name = course.name
+    } catch (error) {
+      if (error instanceof HTTPError) {
+        logger.error(`Received unusual status code ${String(error.response.statusCode)}`)
+        logger.error(`Response body: ${JSON.stringify(error.response.body)}`)
+      } else {
+        logger.error(`An error occurred while making a request to Canvas: ${JSON.stringify(error)}`)
```

Comment:
I feel you should log `error.message` and `error.code` if not `HTTPError`. Incase of undefined status code make it 500
![Screen Shot 2021-06-16 at 3 51 05 PM](https://user-images.githubusercontent.com/8579775/122283606-b7ee0380-ceba-11eb-94c0-5c7b3bef4a92.png)

Reply:
Yeah, so the error object is `any` if it's not an `HTTPError`, so I'd have to check whether `message` and `code` are defined (and they may not be). This is why I'm just stringifying and logging the object. We can return a `500` here to the controller as well, but I guess I'd prefer this more generic logging than a bunch of conditionals checking for defined properties.

Reply:
Not just for logging I also want same think response object that we send to the user as well. We should always send the error responses sent from the upstream source.

Reply:
So if it's not an `HTTPError`, it's likely one of these other ones (https://github.com/sindresorhus/got#gotrequesterror), or an error thrown by our application. Those won't necessarily have status codes associated with them (as they aren't HTTP errors) and may not be related to Canvas.

Reply:
Please see latest commits. I am returning a 500 error if it's not an `HTTPError` with a generic message. I don't anticipate this coming up very often, and we can assess the reporting strategy if/when we start seeing those errors.